dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/communityManager/DataRequestUpdaterTest.kt

```diff
@@ -0,0 +1,87 @@
+import org.dataland.e2etests.auth.JwtAuthenticationHelper
+import org.dataland.e2etests.auth.TechnicalUser
+import org.dataland.e2etests.utils.ApiAccessor
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import java.util.*
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DataRequestUpdaterTest {
+    val apiAccessor = ApiAccessor()
+    val jwtHelper = JwtAuthenticationHelper()
+    private val requestControllerApi = RequestControllerApi(BASE_PATH_TO_COMMUNITY_MANAGER)
+    private val dataController = apiAccessor.dataControllerApiForEuTaxonomyNonFinancials
+    private lateinit var dummyCompanyAssociatedData: CompanyAssociatedDataEutaxonomyNonFinancialsData
+    private val testDataEuTaxonomyNonFinancials = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getTData(1).first()
+
+    private val testCompanyInformation = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getCompanyInformationWithoutIdentifiers(1).first()
+
+    @BeforeAll
+    fun authenticateAsReader() { jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader) }
+
+    @Test
+    fun `post single data request and provide data approve it and change the request status`() {
+        val mapOfIds = apiAccessor.uploadOneCompanyAndEuTaxonomyDataForNonFinancials(
+            testCompanyInformation,
+            testDataEuTaxonomyNonFinancials,
+        )
+
+        val singleDataRequest = SingleDataRequest(
+            companyIdentifier = mapOfIds["companyId"].toString(),
+            frameworkName = SingleDataRequest.FrameworkName.eutaxonomyMinusNonMinusFinancials,
+            listOfReportingPeriods = listOf("2022", "2023"),
+            contactList = listOf("someContact@webserver.de", "simpleString"),
+            message = "This is a test. The current timestamp is ${System.currentTimeMillis()}",
+        )
+
+        val allStoredDataRequests = requestControllerApi.postSingleDataRequest(singleDataRequest)
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        for (storedDataRequest in allStoredDataRequests) {
+            val retrievedDataRequest = requestControllerApi.getDataRequestById(
+                UUID.fromString(storedDataRequest.dataRequestId),
+            )
+            Assertions.assertEquals(storedDataRequest.requestStatus, retrievedDataRequest.requestStatus)
+        }
+
+        uploadDatasetAndValidatePendingState(mapOfIds)
+        Thread.sleep(1000)
+        for (storedDataRequest in allStoredDataRequests) {
+            val retrievedDataRequest = requestControllerApi.getDataRequestById(
+                UUID.fromString(storedDataRequest.dataRequestId),
+            )
+            checkRequestStatusAfterUpload(retrievedDataRequest)
+        }
+    }
+    private fun uploadDatasetAndValidatePendingState(mapOfIds: Map<String, String>) {
```

Comment:
According to the name of the function, I would expect some validation being done inside?

Reply:
Changed the name. Is validation required?

Reply:
I wouldn't say so, especially because we deliberately want to skip QA. Different question: What happens if a dataset is pending review - is the message on the queue just sent once the dataset is approved or can it happen that a request is turned to "answered" before this is done?

Reply:
Nope. The message is only sent on approval or rejection in any of the qa services. However, the request can manually be changed to "answered" via endpopint before by a user.

Reply:
Okay, this is fine. In the community manager nothing happens on rejection, correct?